src/monodroid/jni/inmemory-assemblies.hh

```diff
@@ -0,0 +1,68 @@
+#ifndef INC_MONODROID_INMEMORY_ASSEMBLIES_H
+#define INC_MONODROID_INMEMORY_ASSEMBLIES_H
+
+#include "jni-wrappers.hh"
+#include <string.h>
+#include <mono/metadata/appdomain.h>
+#include <mono/metadata/assembly.h>
+
+#define DEFAULT_CAPACITY 8
+
+namespace xamarin::android::internal {
+	class InMemoryAssemblies
+	{
+		private:
+			struct InMemoryAssemblyEntry
+			{
+				int domain_id;
+				unsigned int assemblies_count;
+				char **names;
+				char **assemblies_bytes;
+				unsigned int *assemblies_bytes_len;
+
+				InMemoryAssemblyEntry (int domain_id, unsigned int assemblies_count, char **names, char **assemblies_bytes, unsigned int *assemblies_bytes_len)
```

Comment:
What I was suggesting on Slack was that this constructor should *itself* perform the `new` operations.  That would make it easier to associate the constructor operations with the the destructor operations, mapping `new T[]` with `delete[]`.